src/syft/grid/duet/__init__.py

```diff
@@ -297,12 +295,23 @@ def join_duet(
     print("♫♫♫ > ...waiting for partner to connect...")
     if loopback:
         loopback_config = {}
-        with open(get_loopback_path(), "r") as f:
-            loopback_config = json.loads(f.read())
-            if "server_id" in loopback_config:
-                target_id = loopback_config["server_id"]
-            else:
-                raise Exception("No loopback file. start the Duet Data Owner first")
+        target_id = ""
+        while target_id == "":
+            try:
```

Comment:
Q: What exception can be thrown here?

Reply:
The file open and json.loads can probably both throw exceptions depending on what happens with the file.